### ISSUE TITLE
Why is Registry find again comparing large Ganga objects behind the scenes?

### ISSUE THREAD

**Issue by rob-c** [ganga-devs/ganga#426]
https://github.com/ganga-devs/ganga/blob/develop/python/Ganga/Core/GangaRepository/Registry.py#L489

I'm slightly concerned that I'm seeing this line in testing. There have been bugs in the ```__eq__``` methods in the past and this is extremely expensive which is why the find operation was at some point rewritten to just compare objects based upon ```_registry_id``` which is assigned at runtime.

This is partly to blame for: http://ganga.web.cern.ch/ganga/release/6.1.19-pre/reports/latest/html/summary_Ganga.old_test.Schema.html

Can we remove this assert as I'm not expecting it to be here?

**Comment by milliams** [comment]
That line was added by you in [29ac12e](https://github.com/ganga-devs/ganga/commit/29ac12e5bf5437528282e7d90a0bb4104673850b#diff-1410a75bf43f8935bb47c3bff87ea0d3R507). I think that really it should be doing an `is` comparison since we're looking for the same actual object.

Overall, that whole function could probably be mostly replaced with:
```python
next(id for id, obj in self._objects.items() if obj is _obj)
```

**Comment by rob-c** [comment]
Quite probably but we need to be certain that this now works. It wouldn't have done in the past due to some peculiarities in reading the XML which I think have gone away now.
Is it worth trying this replacement in a PR?

**Comment by milliams** [comment]
Replacing the function with:
```python
def find(self, _obj):
    """Returns the id of the given object in this registry, or 
    Raise ObjectNotInRegistryError if the Object is not found"""

    obj = stripProxy(_obj)
    try:
        return next(id for id, o in self._objects.items() if o is obj)
    except StopIteration:
        raise ObjectNotInRegistryError("Object '%s' does not seem to be in this registry: %s !" % (getName(obj), self.name))
```
has managed to run through the full test suite locally without problems.

There's no object comparison or even any object inspection occurring so it should be safe. Should I put in a PR for this against `release/6.1.19`?

**Comment by rob-c** [comment]
I'm very happy with this. The ```_registry_id``` checks and such here they were designed to replace the ```==``` in the short-term which can be better replaced with ```is``` now that we're in a better position for managing jobs.
I'd say go for a PR for 6.1.19 to see if it fixes the Schema test in the old config.

I'm seeing some minor problems with prepared state in the LHCb tests again. I'll run some LHCb submission tests myself to see if there are any regressions and report back.

**Comment by rob-c** [comment]
Fixed by #427 but for some reason github wouldn't link the issue and PR
